plugins/decl_hdf5/dataset_op.cxx

```diff
@@ -360,20 +361,68 @@ void Dataset_op::do_write(Context& ctx, hid_t h5_file, hid_t write_lst, const un
 	ctx.logger().trace("Applying `{}' memory selection", dataset_name);
 	m_memory_selection.apply(ctx, h5_mem_space);
 
-	auto&& dataset_type_iter = dsets.find(dataset_name);
 	Datatype_sptr dataset_type;
 	Raii_hid h5_file_type, h5_file_space;
-	if (dataset_type_iter != dsets.end()) {
-		dataset_type = dataset_type_iter->second->evaluate(ctx);
-		tie(h5_file_space, h5_file_type) = space(dataset_type);
-		ctx.logger().trace("Applying `{}' dataset selection", dataset_name);
-		m_dataset_selection.apply(ctx, h5_file_space, h5_mem_space);
+
+	int counter_dataset_found = 0;
+	ctx.logger().trace("search `{}' in the list of datasets section", dataset_name);
+
+	for (auto&& dsets_elem: dsets) {
+		// create regex from string
+		std::regex dsets_elem_regex(dsets_elem.first);
+		// try if dataset_name is including in regex
+		if (std::regex_match(dataset_name, dsets_elem_regex)) {
+			counter_dataset_found++;
```

Comment:
you could just use a boolean:
* if it was already found, trigger an error
* otherwise, set found to true and store the associated type.

Reply:
I agree this makes a better error.